### DISCUSSION TITLE
Standardize case-sensitiveness in Sigma Regexes

### DISCUSSION THREAD

**Discussion by Res260** [SigmaHQ/sigma-specification#100] [General]
The Sigma specification mentions that :

> All values are treated as case-insensitive strings

and

> Regular expressions are case-sensitive by default

However, those are the only two mentions of case-sensitiveness in the whole Sigma specification.
This leads to the following problem: **There is no _standard_ way to express a case-insensitive regex in a Sigma rule.**

From my understanding, this is undefined behavior, which Sigma should try to avoid. Individual backends must decide without guidance how to handle those, which undermines the vendor-agnosticness that we love of Sigma .

For example, in QRadar and splunk, the way to define a case-insensitive regex is by prepending it with `(?i)`. With Crowdstrike Falcon Logscale, however, it's `/myregex/i`. In sqlite, you need to `lower()` the field you're searching on. Completely different ways to express the same outcome. I think Sigma should standardize regex case-sensitiveness, and maybe other regex flags as well.

**Comment by thomaspatzke** [comment]
The [v2 specification](https://github.com/SigmaHQ/sigma-specification/tree/version_2) defines clear ways for case-(in)sensitivity:

* Plain strings can switched to case-sensitive matching with the [`cased` modifier](https://github.com/SigmaHQ/sigma-specification/blob/version_2/appendix_modifer.md#general).
* Regular expression matching can be switched to case-insensitive matching with the [`ignorecase` modifier or its `i` shortcut](https://github.com/SigmaHQ/sigma-specification/blob/version_2/appendix_modifer.md#string-only).

Both modifiers are full supported by pySigma and case-insensitivity syntaxes mentioned above can easily be implemented by backends:

* [Plain case-sensitive match](https://github.com/SigmaHQ/pySigma/blob/ae7c544ee97a6ff9121c9099827b9a44d66f52df/sigma/backends/test/backend.py#L57) with [test as example](https://github.com/SigmaHQ/pySigma/blob/ae7c544ee97a6ff9121c9099827b9a44d66f52df/tests/test_conversion_base.py#L150).
* by default case-insensitive regular expression matching with `(?i)` is implemented [by the base backend class](https://github.com/SigmaHQ/pySigma/blob/ae7c544ee97a6ff9121c9099827b9a44d66f52df/sigma/conversion/base.py#L621) and [works without modification of the backend](https://github.com/SigmaHQ/pySigma/blob/ae7c544ee97a6ff9121c9099827b9a44d66f52df/tests/test_conversion_base.py#L802).
* Other case-sensitive re matching expressions like `/.../i` can be defined by backends as done [in this test](https://github.com/SigmaHQ/pySigma/blob/ae7c544ee97a6ff9121c9099827b9a44d66f52df/tests/test_conversion_base.py#L849).

**Reply by Res260** [reply]
Welp, looks like i'm late :p 

Why is the version 2 still not the main version? What else needs to be defined?

I'll adapt my backend with these specs :)

**Reply by nasbench** [reply]
Version is still a work in progress. I.e things can change

But mainly most new things are defined in there or are still pending review, such as the latest updates around correlations.
